### DISCUSSION TITLE
Custom Payment Gateway Registration Changes

### DISCUSSION THREAD

**Discussion by szagynaf** [impress-org/givewp#6569] [Developer Q&A]
Hi,
I am encouraged by GiveWP Support to ask my question here.

I work at a foundation as a webdeveloper and we are creating our new WP site.
We would like to collect donations at it with Give. However we need a payment gateway for it which is not there. 
So, we are developing one but that payment has a sataus which is different from Give's defaults. 
Before the Give version 2.19.6 there was a hook 'give_payment_statuses' which made possible to register our
own payment status. But since that version this hook is not working. Since this status  is not required 
to be recorded on our site to appropiate serve we can be without it.

Based on this we have a question.
Is it possible that the necessary hooks for payment gateway registration will be removed from the Give durring the developement?

This was my original question. But since that I have read here in an earlier post about the new gateway api 
which has been developed since version 2.18.0 and takes easier to create a custom payment gateway.

So, I would like to complete my question above.

I have'nt found any documentation about that. Can I somewhere?
Is it possible that this api will fully replace the earlier practice?

Thank you!

**Comment by kjohnson** [comment]
@ szagynaf our intention is that the new Gateway API be a more intentional approach to registering payment gateways in GiveWP and all gateway integrations that we maintain are being updated to use this new API over time.

Re: Custom Status: I would first ask how you are planning to use a custom status, which is something that we are looking to move away from with the Gateway API. How are you intending to use the custom status? I may be able to provide a better solution to meet your needs.
